core/unit_test/TestTeamBasic.hpp

```diff
@@ -114,6 +114,47 @@ TEST(TEST_CATEGORY, team_reduce_large) {
   }
 }
 
+template <typename ExecutionSpace>
+struct LargeTeamScratchFunctor {
+  using team_member = typename Kokkos::TeamPolicy<ExecutionSpace>::member_type;
+  const size_t m_per_team_bytes;
+
+  KOKKOS_FUNCTION void operator()(const team_member& member) const {
+    double* team_shared = static_cast<double*>(
+        member.team_scratch(/*level*/ 1).get_shmem(m_per_team_bytes));
+    if (team_shared == nullptr)
+      Kokkos::abort("Couldn't allocate required size!\n");
+    double* team_shared_1 = static_cast<double*>(
+        member.team_scratch(/*level*/ 1).get_shmem(sizeof(double)));
+    if (team_shared_1 != nullptr)
+      Kokkos::abort("Allocated more memory than requested!\n");
+  }
+};
+
+TEST(TEST_CATEGORY, large_team_scratch_size) {
+  const int level   = 1;
+  const int n_teams = 1;
+
+#ifdef KOKKOS_ENABLE_OPENMPTARGET
+  const size_t per_team_extent = 268435460;
+#else
+  const size_t per_team_extent = 502795560;
+#endif
```

Comment:
How did you pick these magic numbers?

Reply:
The second pone was in the issue fixed in the pull request, the first one was chosen such that I allocate more than INT_MAX (2^31-1) bytes. I saw some more problems when choosing the larger size for OpenMPTarget (couldn't allocate enough memory) but was able to trigger the expected problems with the smaller size, too.

Reply:
Please add a comment to that effect